singlePost.php

```diff
@@ -39,5 +42,55 @@
         </div>
     </article>
 </section>
+
+<?php
+if(isset($_SESSION['userid']))
+{
+    ?>
+    <section class="container md:w-1/2 mx-auto mt-5">
+        <form class="p-8 border border-solid rounded-md bg-slate-200" method="post">
+            <div class="mb-5">
+                <label class="mb-3 block" for="content">Comment:</label>
+                <textarea class="w-full" id="content" name="content" rows="5"></textarea>
+            </div>
+            <?php
+            if(isset($_POST['content']))
+            {
+                // inputed special characters won't be interpreted as HTML tags/entities.
+                $content = htmlspecialchars($_POST['content'], ENT_QUOTES, 'UTF-8');
+                $length = strlen($content);
+                if($length < 10 || $length > 200)
+                {
+                    echo "<div style='display: flex; align-items: center; margin-top: -5px; color: red'><p>Comment must be between 10 and 200 characters.</p></div>";
+                } else
+                {
+                    $authorid = $_SESSION['userid'];
+                    $blogid = $_GET['id'];
+                    $timestamp = date('Y-m-d H:i:s');
+                    $commentModel->addComment($authorid, $blogid , $content, $timestamp);
+                    echo "<div style='display: flex; align-items: center; margin-top: -5px; color: green'><p>Comment added successfully!</p></div>";
+                }
+            }
+            ?>
```

Comment:
Can this code move please, it's better to keep heavy logic like this at the top of the file. 

You will of course need to make sure the error messages are displaying in the correct place though.

By putting this code before `$comments = $commentModel->getAllComments($_GET['id']);` you'll find that the comment loads straight away as well, which is nice!